build-caching-staging-1/cloud-init.bash

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+# vim:filetype=sh
+set -o errexit
+set -o pipefail
+
+main() {
+  [[ "${QUIET}" ]] || set -o xtrace
+
+  : "${DEV:=/dev}"
+  : "${ETCDIR:=/etc}"
+  : "${RUNDIR:=/var/tmp/travis-run.d}"
+  : "${VARLIBDIR:=/var/lib}"
+  : "${VARLOGDIR:=/var/log}"
+  : "${VARTMP:=/var/tmp}"
+
+  export DEBIAN_FRONTEND=noninteractive
+  chown nobody:nogroup "${VARTMP}"
+  chmod 0777 "${VARTMP}"
+
+  for substep in \
+    tfw \
+    squignix; do
+    logger running setup substep="${substep}"
+    "__setup_${substep}"
+  done
+}
+
+__wait_for_docker() {
+  local i=0
+
+  while ! docker version; do
+    if [[ $i -gt 600 ]]; then
```

Comment:
Just curious--what's this about? (In what cases would `docker version` exit higher than `600`?)

Reply:
@soulshake Ah, that's the 10m timeout counter comparison, not an exit code check 😁

Reply:
LOL sorry, reading FTW